server/src/main/java/org/elasticsearch/index/mapper/RuntimeFieldType.java

```diff
@@ -39,15 +63,194 @@ public final XContentBuilder toXContent(XContentBuilder builder, Params params)
     /**
      * Prints out the parameters that subclasses expose
      */
-    protected abstract void doXContentBody(XContentBuilder builder, boolean includeDefaults) throws IOException;
+    final void doXContentBody(XContentBuilder builder, boolean includeDefaults) throws IOException {
+        toXContent.accept(builder, includeDefaults);
+    }
+
+    @Override
+    public final boolean isSearchable() {
+        return true;
+    }
+
+    @Override
+    public final boolean isAggregatable() {
+        return true;
```

Comment:
This is the case for all current implementations but not necessarily for all future ones? I think `isSearchable` and `isAggregatable` should probably live in the same class as the concrete methods that implement searching and fielddata

Reply:
I do think that the point of runtime fields is that they are always searchable and aggregatable as they are calculated/loaded at runtime. I cannot think of a runtime fields that would not be searchable and aggregatable... can you?

Reply:
I am leaving this as-is for now, always happy to revisit if needed.